From a9d2f0eed00edbce5af776756e823673cb3fdb06 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Fri, 4 May 2018 17:56:45 +0800
Subject: [PATCH 3727/5242] MLK-18205-8 ARM64: dts: freescale: imx8mm: add
 cpu-idle support

commit  2535028723c6d9f0bb05f8d0fe107b9e678f3406 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add i.MX8MM cpu-idle support, level #1 is wfi,
level #2 is ARM power gated.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi |   16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index 7f90c07..7d9dae2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -45,6 +45,22 @@
 		gpio4 = &gpio5;
 	};
 
+	cpus {
+		idle-states {
+			entry-method = "psci";
+
+			CPU_SLEEP: cpu-sleep {
+				compatible = "arm,idle-state";
+				arm,psci-suspend-param = <0x0010033>;
+				local-timer-stop;
+				entry-latency-us = <1000>;
+				exit-latency-us = <700>;
+				min-residency-us = <2700>;
+				wakeup-latency-us = <1500>;
+			};
+		};
+	};
+
 	memory@40000000 {
 		device_type = "memory";
 		reg = <0x0 0x40000000 0 0x80000000>;
-- 
1.7.9.5

